WordPress/Classes/ViewRelated/Tools/SettingsTextViewController.m

```diff
@@ -172,22 +174,52 @@ - (void)startListeningTextfieldChanges
 {
     // Hook up to Change Events
     [_textField addTarget:self action:@selector(validateTextInput:) forControlEvents:UIControlEventEditingChanged];
-    
-    // Fire initial status
-    [self validateTextInput:_textField];
 }
 
 - (BOOL)textPassesValidation
 {
-    BOOL isEmail = (self.mode == SettingsTextModesEmail);
-    return (self.validatesInput == false || isEmail == false || (isEmail && self.textField.text.isValidEmail));
+    switch(self.mode){
+        case SettingsTextModesEmail:
+            return !self.validatesInput || self.textField.text.isValidEmail;
+            break;
+        case SettingsTextModesLowerCaseText:
+        case SettingsTextModesText:
+            return !self.validatesInput || [self isValidText:self.textField.text];
+            break;
+        case SettingsTextModesURL:
+        case SettingsTextModesPassword:
+        case SettingsTextModesNewPassword:
+        default:
+            return (!self.validatesInput);
+            break;
+    }
+}
+//TODO: This method should be stay in the WorpressShared project
+- (BOOL)isValidText:(NSString*)text
```

Comment:
We have a method in called [`slugForTopicName`](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/develop/WordPressKit/ReaderTopicServiceRemote.m#L364) which contains some of the parsing done before sending this request. It could be a good idea to reuse that here for validation, but I think it's fine to just check the length for now.